src/main/java/org/opensearch/flowframework/rest/RestGetWorkflowStepAction.java

```diff
@@ -69,13 +74,13 @@ protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient
                 );
             }
 
-            ActionRequest request = new ActionRequest() {
-                @Override
-                public ActionRequestValidationException validate() {
-                    return null;
-                }
-            };
-            return channel -> client.execute(GetWorkflowStepAction.INSTANCE, request, ActionListener.wrap(response -> {
+            Map<String, String> params = new HashMap<>();
+            for (String step : steps) {
+                params.put(step, WORKFLOW_STEP);
```

Comment:
While this works, it seems unusual to do the parsing on this side of transport, incurring the transport overhead of the map size and multiple repetitions of the "workflow_step" string (with its length).

Why not just fetch the param as-is and pass the comma-delimited string as a single parameter?  Something like
```java
Map<String, String> params = request.hasParam(WORKFLOW_STEP) 
    ? Map.of(WORKFLOW_STEP, request.param(WORKFLOW_STEP)) 
    : Collections.emptyMap();
```
Then later when you want the list:
```diff
- List<String> steps = new ArrayList<>(request.getParams().keySet());
+ List<String> steps = request.containsKey(WORKFLOW_STEP)
+     ? Arrays.asList(Strings.splitStringByCommaToArray(request.getParams().get(WORKFLOW_STEP)))
+     : Collections.emptyList();
```

Reply:
Better idea!